src/main/java/com/exactpro/th2/act/framework/builders/TableSearchBuilder.java

```diff
@@ -0,0 +1,142 @@
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Iterator;
+import java.util.Map;
+
+import static org.apache.commons.lang3.StringUtils.isEmpty;
+
+public class TableSearchBuilder extends DefaultBuilder<TableSearchBuilder> {
+
+	private Map<String, String> searchParams;
+	private String columnName;
+	private int columnIndex;
+	private int firstRowIndex;
+	private String rowNameFormat;
+	private String rowElementNameFormat;
+	private String rowElementValueFormat;
+
+
+	public TableSearchBuilder(UIFrameworkContext context)
+	{
+		super(context);
+	}
+
+	public TableSearchBuilder searchParams(Map<String, String> searchParams) {
+		this.searchParams = searchParams;
+		return this;
+	}
+
+	public TableSearchBuilder columnName(String columnName) {
+		this.columnName = columnName;
+		return this;
+	}
+
+	public TableSearchBuilder columnIndex(int columnIndex) {
+		this.columnIndex = columnIndex;
+		return this;
+	}
+
+	public TableSearchBuilder setFirstRowIndex(int firstRowIndex)
+	{
+		this.firstRowIndex = firstRowIndex;
+		return this;
+	}
+
+	public TableSearchBuilder setRowNameFormat(String rowNameFormat)
+	{
+		this.rowNameFormat = rowNameFormat;
+		return this;
+	}
+
+	public TableSearchBuilder setRowElementNameFormat(String rowElementNameFormat)
+	{
+		this.rowElementNameFormat = rowElementNameFormat;
+		return this;
+	}
+
+	public TableSearchBuilder setRowElementValueFormat(String rowElementValueFormat)
+	{
+		this.rowElementValueFormat = rowElementValueFormat;
+		return this;
+	}
+
+	@Override
+	protected TableSearchBuilder getBuilder()
+	{
+		return this;
+	}
+
+	@Override
+	protected String getActionName()
+	{
+		return "TableSearch";
+	}
+
+	@Override
+	protected RhAction buildAction() throws UIFrameworkBuildingException
+	{
+		validateParams();
+
+		WinTableSearch.Builder builder = WinTableSearch.newBuilder();
+		builder.addAllLocators(buildWinLocator(winLocator));
+		builder.setId(id);
+		addIfNotEmpty(execute, builder::setExecute);
+		addIfNotEmpty(String.valueOf(firstRowIndex), builder::setFirstRowIndex);
+		addIfNotEmpty(rowNameFormat, builder::setRowNameFormat);
+		addIfNotEmpty(rowElementNameFormat, builder::setRowElementNameFormat);
+		addIfNotEmpty(rowElementValueFormat, builder::setRowElementValueFormat);
+		builder.setSearchParams(createFilters(searchParams));
+		builder.setTargetColumn(columnName);
+		builder.setColumnIndex(String.valueOf(columnIndex));
+
+		return RhAction.newBuilder().setWinTableSearch(builder.build()).build();
+	}
+
+	private void validateParams() throws UIFrameworkBuildingException {
+		if (searchParams == null || searchParams.isEmpty() || isEmpty(columnName))
+			throw new UIFrameworkBuildingException("Search params or column name cannot be empty");
+
+		if (columnIndex < 0)
+			throw new UIFrameworkBuildingException("Column index cannot be negative");
+		
+		if (StringUtils.isBlank(id))
+			throw new UIFrameworkBuildingException("'Id' cannot be empty");
```

Comment:
throw new UIFrameworkBuildingException("Id cannot be empty");

because other parameters are printed without quotes.